src/DGEList.py

```diff
@@ -223,15 +259,44 @@ def tpm(
         # TODO: Implement here
 
         # self = self.cpm(log=log, prior_count=prior_count)
-        return self
 
     def __repr__(self) -> str:
         """Give a pretty non-executeable representation of this object."""
-        num_samples = len(self.samples) if self.samples is not None else 0
-        num_genes = len(self.genes) if self.genes is not None else 0
+        num_samples = len(self._samples) if self._samples is not None else 0
+        num_genes = len(self._genes) if self._genes is not None else 0
```

Comment:
These should access the public properties, not private.

Reply:
Why is that?  I don't want to use the accessors, just check the length of the private objects, no?